Esri/arcgis-maps-sdk-swift-samples
# [New] Filter by definition expression or display filter

## Description

This PR implements `Filter by definition expression or display filter` in `Search and Query` category.
URL to README: [README](https://github.com/Esri/arcgis-maps-sdk-swift-samples/tree/Caleb/New-FilterByDefinitionExpressionOrDisplayFilter/Shared/Samples/Filter%20by%20definition%20expression%20or%20display%20filter)

## Linked Issue(s)

- `swift/issues/5369`

## How To Test

- Tap "Expression" to filter the feature layer with the definition expression.
- Tap "Display Filter" to filter the results with the display filter definition.
- Tap "None" to view feature layer without filtering.
- See the feature count change when you pan, zoom, and update the filtering selection.

## Screenshots

![filter-by-definition-expression-or-display-filter](https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/34120440/43bbc0c5-6635-4f00-b0b4-67d1bfc6196b)
